File: NET/UniversitySchedule.Core/Coordinates.cs
using System;
using System.Diagnostics;
using System.Runtime.Serialization;

namespace Mntone.UniversitySchedule.Core
{
	/// <summary>
	/// Coordinates
	/// </summary>
	[DataContract]
	[DebuggerDisplay( "{StringView()}" )]
	public sealed class Coordinates
	{
		private Coordinates() { }

		/// <summary>
		/// Constructor
		/// </summary>
		/// <param name="latitude">Latitude</param>
		/// <param name="longitude">Longitude</param>
		internal Coordinates( float latitude, float longitude )
		{
			this.Latitude = latitude;
			this.Longitude = longitude;
		}

		/// <summary>
		/// Latitude
		/// </summary>
		[DataMember( Name = "latitude", IsRequired = true )]
		public float Latitude { get; private set; }

		/// <summary>
		/// Longitude
		/// </summary>
		[DataMember( Name = "longitude", IsRequired = true )]
		public float Longitude { get; private set; }

		private string StringView()
		{
			var absLatitude = Math.Abs( this.Latitude );
			var absLongitude = Math.Abs( this.Longitude );
			var latitudeDegrees = ( uint )absLatitude;
			var longitudeDegrees = ( uint )absLongitude;
			var latitudeMinutesF = 60 * ( absLatitude - latitudeDegrees );
			var longitudeMinutesF = 60 * ( absLongitude - longitudeDegrees );
			var latitudeMinutes = ( uint )latitudeMinutesF;
			var longitudeMinutes = ( uint )longitudeMinutesF;
			var latitudeSeconds = 60 * ( latitudeMinutesF - latitudeMinutes );
			var longitudeSeconds = 60 * ( longitudeMinutesF - longitudeMinutes );
			var charLatitude = this.Latitude > 0 ? 'N' : 'S';
			var charLongitude = this.Latitude > 0 ? 'E' : 'W';
			return string.Format(
				"{0}°{1}'{2:0.000}\"{3}, {4}°{5}'{6:0.000}\"{7}",
				latitudeDegrees, latitudeMinutes, latitudeSeconds, charLatitude,
				longitudeDegrees, longitudeMinutes, longitudeSeconds, charLongitude );
		}
	}
}

File: JVM/univschedule-core/src/main/java/mntone/univschedule/core/Coordinates.java
package mntone.univschedule.core;

import org.json.JSONObject;

/**
 * Coordinates
 */
public final class Coordinates
{
	private final float mLatitude;
	private final float mLongitude;

	/**
	 * Initialize a new Coordinates.
	 *
	 * @param latitude the latitude
	 * @param longitude the longitude
	 */
	Coordinates( final float latitude, final float longitude )
	{
		this.mLatitude = latitude;
		this.mLongitude = longitude;
	}

	/**
	 * Initialize a new Coordinates.
	 *
	 * @param coordinates the coordinates
	 */
	Coordinates( final JSONObject coordinates )
	{
		this.mLatitude = ( float )coordinates.getDouble( "latitude" );
		this.mLongitude = ( float )coordinates.getDouble( "longitude" );
	}

	/**
	 * Get latitude.
	 *
	 * @return the latitude
	 */
	public float getLatitude()
	{
		return this.mLatitude;
	}

	/**
	 * Get longitude.
	 *
	 * @return the longitude
	 */
	public float getLongitude()
	{
		return this.mLongitude;
	}
}